api/src/main/java/com/redhat/ipaas/api/v1/rest/Lister.java

```diff
@@ -39,11 +39,19 @@
             paramType = "query", dataType = "string"),
         @ApiImplicitParam(
             name = "direction", value = "Sorting direction when a 'sort' field is provided. Can be 'asc' " +
-            "(ascending) or 'desc' (descending)", paramType = "query", dataType = "string")
+            "(ascending) or 'desc' (descending)", paramType = "query", dataType = "string"),
+        @ApiImplicitParam(
+            name = "page", value = "Page number to return", paramType = "query", dataType = "integer", defaultValue = "1"),
+        @ApiImplicitParam(
+            name = "per_page", value = "Number of records per page", paramType = "query", dataType = "integer", defaultValue = "20")
 
     })
-    default Collection<T> list(@Context UriInfo uriInfo) {
-        return getDataManager().fetchAll(resourceKind(), new ReflectiveSorter<>(resourceClass(), new SortOptionsFromQueryParams(uriInfo)));
+    default ListResult<T> list(@Context UriInfo uriInfo) {
+        List<T> fetched = getDataManager().fetchAll(resourceKind(), new ReflectiveSorter<>(resourceClass(), new SortOptionsFromQueryParams(uriInfo)));
+        return new ListResult.Builder<T>()
+            .totalCount(fetched.size())
```

Comment:
Total size of all results to allow the client to know when to do next/previous paging.